provider/pkg/provider/remote/commandOutputs.go

```diff
@@ -40,7 +40,10 @@ func (c *CommandOutputs) run(ctx p.Context, cmd string) error {
 
 	if c.Environment != nil {
 		for k, v := range *c.Environment {
```

Comment:
The `CommandInputs.Environments` field shouldn't need to be a `*map[string]string`, since `map[string]string` is nillable. The change is out of scope, but I would recommend changing it to improve the authoring experience.

```suggestion
		for k, v := range c.Environment {
```